packages/shared/src/permissions/permissionsClass.ts

```diff
@@ -643,6 +643,48 @@ export class Permissions {
     );
   }
 
+  public canReadData = (projects: string | string[] | undefined): boolean => {
```

Comment:
I think I've found an issue with this.

With a lot of our resources, the `projects` array property is optional, so in the event it's undefined,  the first code block in this function won't evaluate as expected, unless when calling `canReadData` manually add an empty array E.G. `canReadData(datasource.projects || [])`. Otherwise, a user with a global role of `noaccess` won't be able to see the data sources in `All Projects`.

Reply:
It feels like we almost need two different `canReadData` helpers - 1 for resources with a `project` property, and one for those with a `projects` property. Not sure how we'd name those right now.

Or, we could update this signature to accept a resource type.

E.G. 
`canReadData = (projects: string | string[] | undefined, resourceType: "datasource" | "experiment" | "metric" ....)`

Then the first block can be:
```ts
if (RESOURCES_WITH_PROJECTS.includes(resourceType) && (!projects.length || projects === undefined)) {
// return true
}
```

A third option, is have a `canReadRESOURCE` for each resource.

Reply:
I would go with `canReadSingleProjectResource(project: string | undefined)` and `canReadMultiProjectResource(projects: string[] | undefined)`